pkg/workload/tpcc/README.md

```diff
@@ -0,0 +1,312 @@
+When using multiple workload generators within a single-region CockroachDB cluster, workload partitioning is essential. 
+This feature divides warehouses among workload nodes, with each node targeting only its assigned warehouses. Without 
+partitioning, nodes would compete for the same warehouses, causing transaction contention and deadlocks that reduce 
+performance.
+
+```bash
+--client-partitions=N    # Number of partitions to divide warehouses into
+--partition-affinity=K   # Specifies which partition(s) this client targets, from 0 to N-1. Multiple values can be specified using commas
+```
+
+Consider a configuration with:
+- Total warehouses: 10,000
+- Active warehouses: 5,000
+- Client partitions: 5
+
+The warehouse partitioning logic is implemented in [`pkg/workload/tpcc/partition.go`](https://github.com/cockroachdb/cockroach/blob/master/pkg/workload/tpcc/partition.go#L274).
+Warehouse distribution across partitions:
+```bash
+Total Warehouse Ranges:
+Partition 0: warehouses [0-2000)      # 2000 warehouses
+Partition 1: warehouses [2000-4000)   # 2000 warehouses
+Partition 2: warehouses [4000-6000)   # 2000 warehouses
+Partition 3: warehouses [6000-8000)   # 2000 warehouses
+Partition 4: warehouses [8000-10000)  # 2000 warehouses
+
+Active Warehouse Ranges (5000 active):
+Partition 0: warehouses [0-1000)      # 1000 active warehouses
+Partition 1: warehouses [2000-3000)   # 1000 active warehouses
+Partition 2: warehouses [4000-5000)   # 1000 active warehouses
+Partition 3: warehouses [6000-7000)   # 1000 active warehouses
+Partition 4: warehouses [8000-9000)   # 1000 active warehouses
+```
+
+### Connection Distribution with Partitioning
+This section explains how connections are distributed when using client-side partitioning. When `--client-partitions`
+and `--partition-affinity` are specified, each client establishes connections only to the warehouses in its assigned partition(s).
+```bash
+--conns=N  # Number of connections. Defaults to --warehouses * 2 (except in nowait mode, where it defaults to --workers)
+```
+Implementation details can be found in [`pkg/workload/tpcc/tpcc.go`](https://github.com/cockroachdb/cockroach/blob/master/pkg/workload/tpcc/tpcc.go#L967).
+
+1. Single Partition Affinity
+```bash
+# Example Configuration
+--client-partitions=3
+--partition-affinity=0
+URLs=[node1:26257, node2:26257, node3:26257]
+
+# Distribution
+Partition 0: [node1, node2, node3]  # All connections assigned to partition 1
+```
+
+2. When multiple partition affinities are specified, connections are mapped one-to-one in order of URLs and affinity partitions.
+```bash
+# Example Configuration
+--partitions=3
+--partition-affinity=0,1
+URLs=[node1:26257, node2:26257, node3:26257]
+
+# Distribution
+Partition 0: [node1]                  # From affinity mapping
+Partition 1: [node2]                  # From affinity mapping
+Partition 2: [node1,node2,node3]      # From fallback, this won't be used as affinity is not specified
```

Comment:
It's not immediately obvious what "fallback" means in this context. I think it's worth explaining what happens when a partition or multiple partitions are _not_ covered by the `--partition-affinity` spec. Also, we should mention that the mapping of partitions to URLs uses round-robin partitioning; maybe also change the example to illustrate this better; e.g., 5 nodes and 2 partitions.